metarecord/views/function.py

```diff
@@ -78,27 +73,52 @@ def get_fields(self):
 
         return fields
 
-    def _create_new_version(self, function_data):
+    def _create_new_version(self, function_data, copy_from_previous=False, phases=[]):
         user = self.context['request'].user
         user_data = {'created_by': user, 'modified_by': user}
 
         phase_data = function_data.pop('phases', [])
+
         function_data.update(user_data)
         function = Function.objects.create(**function_data)
 
-        for index, phase_datum in enumerate(phase_data, 1):
-            action_data = phase_datum.pop('actions', [])
-            phase_datum.update(user_data)
-            phase = Phase.objects.create(function=function, index=index, **phase_datum)
+        if copy_from_previous:
+            for phase in phases:
+                actions = []
+                for action in phase.actions.all():
+                    records = []
+                    for record in action.records.all():
+                        record.pk = None
+                        record.save()
+                        records.append(record)
+
+                    action.pk = None
+                    action.save()
+                    for record in records:
+                        action.records.add(record)
+                    actions.append(action)
+
+                phase.pk = None
+                phase.save()
+                for action in actions:
+                    phase.actions.add(action)
+                function.phases.add(phase)
```

Comment:
Follow up to the previous comment. Phases, actions and records should be assigned newly generated uuid v4 before saving.